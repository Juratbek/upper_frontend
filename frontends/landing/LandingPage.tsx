import { Divider } from 'components/lib';
import { Labels, PublishedArticle, TopLabel } from 'components/molecules';
import { Fragment } from 'react';
import { usePublishedArticlesList } from 'store/clients/published-article';
import { addAmazonBucketUrl } from 'utils/published-article';

import { BlueBox } from './blue-box/BlueBox';
import { LandingHeader } from './header/LandingHeader';

export const LandingPage = (): JSX.Element => {
  const { list: articles } = usePublishedArticlesList(TopLabel);
  return (
    <div>
      <LandingHeader />
      <main className='container' style={{ paddingTop: 48 }}>
        <BlueBox />
        <section style={{ maxWidth: 720 }}>
          <Labels />
          {articles.map(addAmazonBucketUrl).map((article) => (
            <Fragment key={article.id}>
              <Divider color='secondary' />
              <PublishedArticle article={article} />
            </Fragment>
          ))}
        </section>
      </main>
    </div>
  );
};
