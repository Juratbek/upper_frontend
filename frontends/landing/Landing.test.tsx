import { render } from '@testing-library/react';
import { QueryWrapper, ReduxWrapper } from 'tests';

import { LandingPage } from './LandingPage';

const Landing = (
  <ReduxWrapper>
    <QueryWrapper>
      <LandingPage />
    </QueryWrapper>
  </ReduxWrapper>
);

describe('Landing page content', () => {
  it('snapshot', () => {
    render(Landing);
    expect(document.body).toMatchSnapshot();
  });
});
