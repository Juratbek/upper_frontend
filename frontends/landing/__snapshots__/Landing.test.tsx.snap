// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Landing page content > snapshot 1`] = `
<body>
  <div>
    <div
      class="root"
    >
      <header
        class="header container"
      >
        <div
          class="search-container hidden"
        >
          <span>
            <svg
              fill="none"
              height="24"
              role="icon"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M15 6L9 12L15 18"
                stroke="black"
                stroke-width="1.6"
              />
            </svg>
          </span>
          <div
            class="position-relative w-100"
          >
            <input
              class="search-input input"
              style="border-color: rgba(0, 0, 0, 0.25);"
            />
          </div>
        </div>
        <a
          class="logo link"
          href="/web"
        >
          <svg
            data-testid="logo-svg"
            fill="none"
            role="img"
            viewBox="0 0 188 80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              fill="white"
              id="path-1-inside-1_804_345"
            >
              <path
                d="M0 0H80V80H0V0Z"
              />
            </mask>
            <path
              d="M0 0V-5H-5V0H0ZM0 80H-5V85H0V80ZM0 5H80V-5H0V5ZM80 75H0V85H80V75ZM5 80V0H-5V80H5Z"
              fill="#54A9EB"
              mask="url(#path-1-inside-1_804_345)"
            />
            <line
              stroke="#54A9EB"
              stroke-width="5"
              x1="77.5"
              x2="77.5"
              y1="5"
              y2="17"
            />
            <line
              stroke="#54A9EB"
              stroke-width="5"
              x1="77.5"
              x2="77.5"
              y1="63"
              y2="75"
            />
            <path
              d="M50.8324 21.9091H55.3239V46.4673C55.3239 49.0028 54.7262 51.2667 53.5309 53.2589C52.3477 55.239 50.6754 56.8026 48.5142 57.9496C46.353 59.0845 43.8175 59.652 40.9077 59.652C37.9979 59.652 35.4624 59.0845 33.3011 57.9496C31.1399 56.8026 29.4616 55.239 28.2663 53.2589C27.0831 51.2667 26.4915 49.0028 26.4915 46.4673V21.9091H30.983V46.1051C30.983 47.9162 31.3814 49.5281 32.1783 50.9407C32.9751 52.3413 34.1101 53.446 35.5831 54.255C37.0682 55.0518 38.843 55.4503 40.9077 55.4503C42.9723 55.4503 44.7472 55.0518 46.2322 54.255C47.7173 53.446 48.8523 52.3413 49.6371 50.9407C50.434 49.5281 50.8324 47.9162 50.8324 46.1051V21.9091Z"
              fill="#54A9EB"
            />
            <path
              d="M64.2932 59V21.9091H76.8259C79.7357 21.9091 82.1143 22.4343 83.9616 23.4847C85.8209 24.5231 87.1974 25.9297 88.0908 27.7045C88.9843 29.4794 89.431 31.4595 89.431 33.6449C89.431 35.8303 88.9843 37.8164 88.0908 39.6033C87.2094 41.3903 85.8451 42.815 83.9978 43.8775C82.1505 44.9279 79.784 45.4531 76.8983 45.4531H67.9154V41.4688H76.7535C78.7457 41.4688 80.3454 41.1246 81.5528 40.4364C82.7602 39.7482 83.6356 38.8185 84.1789 37.6474C84.7343 36.4641 85.012 35.13 85.012 33.6449C85.012 32.1598 84.7343 30.8317 84.1789 29.6605C83.6356 28.4893 82.7542 27.5717 81.5347 26.9077C80.3153 26.2315 78.6974 25.8935 76.681 25.8935H68.7847V59H64.2932ZM96.6663 59V21.9091H109.199C112.109 21.9091 114.487 22.4343 116.335 23.4847C118.194 24.5231 119.57 25.9297 120.464 27.7045C121.357 29.4794 121.804 31.4595 121.804 33.6449C121.804 35.8303 121.357 37.8164 120.464 39.6033C119.582 41.3903 118.218 42.815 116.371 43.8775C114.524 44.9279 112.157 45.4531 109.271 45.4531H100.288V41.4688H109.127C111.119 41.4688 112.718 41.1246 113.926 40.4364C115.133 39.7482 116.009 38.8185 116.552 37.6474C117.107 36.4641 117.385 35.13 117.385 33.6449C117.385 32.1598 117.107 30.8317 116.552 29.6605C116.009 28.4893 115.127 27.5717 113.908 26.9077C112.688 26.2315 111.07 25.8935 109.054 25.8935H101.158V59H96.6663ZM129.039 59V21.9091H151.424V25.8935H133.531V38.4261H150.265V42.4105H133.531V55.0156H151.714V59H129.039ZM159.52 59V21.9091H172.052C174.95 21.9091 177.329 22.4041 179.188 23.3942C181.047 24.3722 182.424 25.7184 183.317 27.4329C184.211 29.1474 184.658 31.0973 184.658 33.2827C184.658 35.468 184.211 37.4059 183.317 39.0962C182.424 40.7866 181.054 42.1147 179.206 43.0806C177.359 44.0344 174.998 44.5114 172.125 44.5114H161.983V40.4545H171.98C173.96 40.4545 175.554 40.1648 176.761 39.5852C177.981 39.0057 178.862 38.1847 179.405 37.1222C179.961 36.0476 180.239 34.7678 180.239 33.2827C180.239 31.7976 179.961 30.4996 179.405 29.3888C178.85 28.2781 177.963 27.4208 176.743 26.8171C175.524 26.2013 173.912 25.8935 171.908 25.8935H164.011V59H159.52ZM176.979 42.3381L186.106 59H180.891L171.908 42.3381H176.979Z"
              fill="black"
            />
          </svg>
        </a>
        <span
          class="search-icon"
        >
          <svg
            fill="none"
            height="24"
            role="icon"
            viewBox="0 0 20 20"
            width="24"
          >
            <path
              d="M17.4344 18.3334C17.204 18.3334 16.9736 18.2437 16.8072 18.0772L14.4262 15.6944C14.0806 15.3485 14.0806 14.7849 14.4262 14.4262C14.7718 14.0803 15.3351 14.0803 15.6935 14.4262L18.0744 16.8089C18.4201 17.1548 18.4201 17.7185 18.0744 18.0772C17.8952 18.2437 17.6648 18.3334 17.4344 18.3334Z"
              fill="black"
              stroke="#F5F5F5"
              stroke-width="0.6"
            />
            <path
              d="M16.8003 9.58341C16.8003 13.5691 13.5693 16.8001 9.58366 16.8001C5.598 16.8001 2.36699 13.5691 2.36699 9.58341C2.36699 5.59776 5.598 2.36675 9.58366 2.36675C13.5693 2.36675 16.8003 5.59776 16.8003 9.58341Z"
              stroke="black"
              stroke-width="1.4"
            />
          </svg>
        </span>
      </header>
    </div>
    <main
      class="container"
      style="padding-top: 48px; flex: 1; width: 100%;"
    >
      <div
        class="container"
      >
        <h1
          class="heading"
        >
          upper bilan yanada yuqoriroq
        </h1>
        <div
          class="body"
        >
          <button
            class="root start"
          >
            <span
              class="start-text"
            >
              Boshladik
            </span>
            <span
              class="up-right-icon"
            >
              <svg
                fill="none"
                height="20"
                role="icon"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M1.5 2H17.5M17.5 2V18M17.5 2L1.5 18"
                  stroke="#000C1A"
                  stroke-width="4"
                />
              </svg>
            </span>
          </button>
          <div
            class="container"
          >
            <img
              alt="vs code"
              class="image"
              data-nimg="1"
              decoding="async"
              height="120"
              loading="lazy"
              src="/_next/image?url=%2Flogos%2Fvs_code.png&w=256&q=75"
              srcset="/_next/image?url=%2Flogos%2Fvs_code.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fvs_code.png&w=256&q=75 2x"
              style="color: transparent; --i: 0;"
              width="120"
            />
            <img
              alt="vs code"
              class="image"
              data-nimg="1"
              decoding="async"
              height="120"
              loading="lazy"
              src="/_next/image?url=%2Flogos%2Fchat-gpt.png&w=256&q=75"
              srcset="/_next/image?url=%2Flogos%2Fchat-gpt.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fchat-gpt.png&w=256&q=75 2x"
              style="color: transparent; --i: 1;"
              width="120"
            />
            <img
              alt="vs code"
              class="image"
              data-nimg="1"
              decoding="async"
              height="120"
              loading="lazy"
              src="/_next/image?url=%2Flogos%2Fjs.png&w=256&q=75"
              srcset="/_next/image?url=%2Flogos%2Fjs.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fjs.png&w=256&q=75 2x"
              style="color: transparent; --i: 2;"
              width="120"
            />
            <img
              alt="vs code"
              class="image"
              data-nimg="1"
              decoding="async"
              height="120"
              loading="lazy"
              src="/_next/image?url=%2Flogos%2Fadobe.png&w=256&q=75"
              srcset="/_next/image?url=%2Flogos%2Fadobe.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fadobe.png&w=256&q=75 2x"
              style="color: transparent; --i: 3;"
              width="120"
            />
          </div>
        </div>
      </div>
      <div
        class="root"
      >
        <section
          class="container"
        >
          <div
            class="root"
            style="padding-top: 2rem; margin-top: -2rem;"
          >
            <div
              class="labels-container"
              id="labels"
            >
              <div
                class="lebels-list"
              >
                <button
                  class="button outlined"
                  color="outlined"
                >
                  Eng so'nggi
                </button>
                <button
                  class="button primary"
                  color="primary"
                >
                  Top
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  Java
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  JavaScript
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  TypeScript
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  HTML
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  C
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  SQL
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  Python
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  C++
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  Go
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  C#
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  PHP
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  Swift
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  Kotlin
                </button>
                <button
                  class="button outlined"
                  color="outlined"
                >
                  Rust
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              class="content-center loading"
            >
              <div
                class="spinner spinner--primary"
                role="spinner"
              />
            </div>
          </div>
        </section>
        <div
          class="sidebar"
          style="top: 2rem;"
        >
          <div
            class="add-container"
          >
            <h4
              class="sale"
            >
              Chegirma
            </h4>
            <h3
              class="percent"
            >
              15%
            </h3>
            <p
              class="text"
            >
              UPPER promokodi bilan Frontint.uz kurslariga chegirma oling
            </p>
          </div>
          <a
            href="https://frontint.uz/"
            rel="noreferrer"
            target="_blank"
          >
            <button
              class="button mt-1 w-100 btn"
              style="opacity: 1;"
            >
              Chegirma olish
            </button>
          </a>
        </div>
      </div>
    </main>
  </div>
</body>
`;
